homeassistant/components/synology_dsm/const.py

```diff
@@ -247,5 +248,26 @@
     },
 }
 
+INFORMATION_SENSORS = {
+    f"{SynoDSMInformation.API_KEY}:temperature": {
+        ENTITY_NAME: "temperature",
+        ENTITY_UNIT: None,
+        ENTITY_ICON: "mdi:thermometer",
+        ENTITY_CLASS: "temperature",
+        ENTITY_ENABLE: True,
+    },
+    f"{SynoDSMInformation.API_KEY}:uptime": {
```

Comment:
Please make this sensor disabled by default as uptime sensors hammer the state machine with updates.

Reply:
https://developers.home-assistant.io/docs/entity_registry_disabled_by/#integrations-setting-default-value-of-disabled_by-for-new-entity-registry-entries

Reply:
this should use the device class of timestamp and do the math to return the start time. example here:
https://github.com/home-assistant/core/blob/dev/homeassistant/components/nzbget/sensor.py#L121

Reply:
@bdraco 
I wasn't aware of `entity_registry_enabled_default` before I saw it during the development, but I still don't fully got it. If set to `False`, does it means it won't never be updated (unless enable) ? What do you mean by hammering the state machine ?

 @ctalkington 
Actually I hesitated before transforming what HA is receiving from the API. I'm also more interested by the last boot then the real uptime in seconds (On my HA, I did a template sensor to transform the raw uptime to the last boot). Anyway, If you propose to transform it directly in the integration, i'm fine with that. I guess the entity should then be named 'last boot' right ?

Reply:
@thomasgermain 

> I wasn't aware of entity_registry_enabled_default before I saw it during the development, but I still don't fully got it. If set to False, does it means it won't never be updated (unless enable) ? What do you mean by hammering the state machine ?

When set to `False` the entity will not be created unless you go and manually enable it
on the integration page.

Since its a timestamp, it is always going to trigger a state changed event every time
the integration fetches new data.  We generally avoid enabling these type of sensors
by default because of performance concerns.

Reply:
it can still be named uptime as thats what its derived from. on the frontend it will just convert the timestamp to human format like relative time

Reply:
the coverted timestamp shouldnt trigger state change unless its been restarted as the number of seconds should always point back to same point in time? the last start. I guess there can be inaccuracies with time conversiom and over nefwork but it should all be pretty reasonable on state machine

Reply:
but i do think uptime type sensors are good candidates for disabled by default eitherway esp if adding them and havent been used by people yet

Reply:
yes, i'm going to set it to  `False`.

I still have question about this uptime sensor, in the example you gave it already returns the formatted date, not a timestamp, this is why I think it could be named `last boot`

Reply:
> @thomasgermain
> 
> > I wasn't aware of entity_registry_enabled_default before I saw it during the development, but I still don't fully got it. If set to False, does it means it won't never be updated (unless enable) ? What do you mean by hammering the state machine ?
> 
> When set to `False` the entity will not be created unless you go and manually enable it
> on the integration page.
> 
> Since its a timestamp, it is always going to trigger a state changed event every time
> the integration fetches new data. We generally avoid enabling these type of sensors
> by default because of performance concerns.

Yes agree on that concern. Especially since these devices are shut off in quite a few setups, to preserve power consumption (during the night/off hours). Would be harming the system if any new sensors would keep trying to update in that case. So, not only thinking of disabling by default, but maybe also consider a system of sensing if the Nas is online in the first place. (compare the Plex sensors, which, when installed on the Synology, keep trying to update and keep throwing warnings and flooding the go with that..)

Reply:
I'm not sure to understand it correctly:
- by harming the system, you mean HA right ? I don't get how HA can be harmed: here is my understanding, the entity state will change and it will trigger a state changed event (only if the state change or every time ?). But this will only happen every 15min, or even more if the state doesn't change
- for the system of sensing, I think it's already there, if the synology replies, it means it's online (or maybe I also don't get that part)

To avoid changing the state of the last boot sensor everytime, maybe I can check if the new value coming from the synology is bigger than the current state. If true, the synology hasn't restarded (then it returns the previous state), otherwise it returns the new state. Does it seems good ?

I expected an easier PR than that :stuck_out_tongue:, but at least, i'm learning something :smiley:

Reply:
Haven't check all the conv but yes:
- `DEVICE_CLASS_TIMESTAMP`
- `None` to unit and icon

You can update all this file like this please.